tests/test_signal_generator.py

```diff
@@ -117,3 +145,93 @@ def test_generate_flmn(rng, L, N, L_lower, reality):
         assert np.allclose(f_complex.imag, 0)
         f_real = s2fft.wigner.inverse(flmn, L, N, reality=True, L_lower=L_lower)
         assert np.allclose(f_complex.real, f_real)
+
+
+def gaussian_covariance(spectra):
+    """Gaussian covariance for a stack of spectra.
+
+    If the shape of *spectra* is *(K, K, L)*, the shape of the
+    covariance is *(L, C, C)*, where ``C = K * (K + 1) // 2``
+    is the number of independent spectra.
+
+    """
+    _, K, L = spectra.shape
+    row, col = np.tril_indices(K)
+    cov = np.zeros((L, row.size, col.size))
+    ell = np.arange(L)
+    for i, j in np.ndindex(row.size, col.size):
+        cov[:, i, j] = (
+            spectra[row[i], row[j]] * spectra[col[i], col[j]]
+            + spectra[row[i], col[j]] * spectra[col[i], row[j]]
+        ) / (2 * ell + 1)
+    return cov
+
+
+@pytest.mark.flaky
```

Comment:
If `rng` is fixed then this test isn't actually flaky, and `pytest-rerunfailures` isn't needed.

Reply:
By

> If `rng` is fixed

do you mean if the `numpy.random.Generator` instance returned by the `rng` fixture has a fixed seed? If so, then yes this already the case - there is a default fixed seed set in `tests/conftests.py` and it can be altered by passing a `--seed SEED` argument to `pytest`.